src/Helpers/OAuth2/OAuthConfig.php

```diff
@@ -224,6 +232,38 @@ public function setDefaultScopes(array $defaultScopes): static
         return $this;
     }
 
+    /**
+     * Set the request modifier callable which can be used to modify the request being sent
+     *
+     * @param callable(\Saloon\Contracts\Request): (void) $requestModifier
+     * @return $this
+     */
+    public function setRequestModifier(callable $requestModifier): static
+    {
+        $this->requestModifier = $requestModifier;
```

Comment:
Turn into a Closure, if we change the property from mixed to Closure.

```suggestion
        $this->requestModifier = $requestModifier(...);
```